auth.php

```diff
@@ -147,4 +192,100 @@ public function user_login ($username, $password) {
         return false;
     }
 
+    /**
+     * Check if the password is master password.
+     * @param $userpassword
+     * @return bool
+     */
+    private function is_master_password($userpassword) {
+        global $CFG;
+        if (isset($CFG->forced_plugin_settings)) {
```

Comment:
Do we need to ensure this feature to be run on Non Production Environment only? (E.g the debug mode is set to "Developer")

Reply:
Yeah I was thinking about ways to detect this. The problem is that debug mode is not always on even in dev. I think forcing it to be set in config.php is enough